### ISSUE TITLE
Upgrade to Vue3

### ISSUE THREAD

**Issue by rotsee** [fruktkartan/fruktkartan#92]
Vue 3 is a major rewrite with a large performance boost, and also Vue2 is really tricky to maintain due to old dependencies that don't play well with newer Node versions.

Might require some work to make the Lefalet implementation work though

**Comment by Greenheart** [comment]
I'm exploring the possibility to rewrite the frontend in https://svelte.dev

This would allow using Leaflet directly since Svelte integrates nicely with regular JS libraries.

There are many advantages, but most importantly is the simplified DX and easier maintenance compared to Vue, while keeping a similar syntax.

**Comment by rotsee** [comment]
There is now a branch with the whole codebase migrated to Vue3: https://github.com/fruktkartan/fruktkartan/tree/vue3

## Major changes

- Snackbar messages handled in a central pinia store (with a simple message queue)
- DayJS (only used in one place) replaced with native JS toLocaleDateString
- Code rewritten using the new and more ergonomic script setup-syntax

## Still TBD

- Prettier temporarily removed as I had issues setting it up (config still left)
- Store last map coordinates in localStorage, like we did before
- Testing, testing, testing

## Next steps

- Nuxt3 has built-in backend support. I've used it before, and it's very pleasant to work with. Allows us to merge this repo with the API repo, run all code on the same server, and also share some business logic (around tree types etc) between backend and frontend
- Routing is a bit of a mess, just as it was before. Haven't touched this

## Next-next steps

- The Vue Leaflet package is abandonware, and also integrates quite poorly into the Vue system. Sooner or later we will want to replace that with something else.
- Having some e2e tests would be nice

(ping @quite )

**Comment by quite** [comment]
Just running `npm ci && npm run serve` (would be great if required node/npm versions could be specified!) results in being served a 404 at the root. Is `public/index.html` missing, or is it no longer needed and something else is wrong?

**Comment by rotsee** [comment]
index.html is no longer needed.

To serve a static build, you must first run `npm run build`

(Or run `npm run dev` to start a dev server)

**Comment by quite** [comment]
Ah ok, `serve` used to run the dev server. I pushed a change to vue3 branch.

**Comment by rotsee** [comment]
@quite At some point I'd like to be able to test this branch when deployed. Could you help me deploy it to Netlify? I'm not sure I have the right permissions, and/or I haven't been able to figure out how our workflow is setup. I've never used Netlify, so it could very well be me not understanding what I'm supposed to do, and where.

Also: Does our Netlify setup handle full stack deploys? I'd like to merge this repo with the backend repo. That's much easier now with Nuxt 3 (but will require that we migrate the database from Heroku).

**Comment by quite** [comment]
I managed to get into Netlify again, and it looks like it's not possible to add more users without paying up (not sure this was always the case). Anyway, I now set Netlify's build image to ubuntu-24 ("Noble") and had it build and deploy the master branch to https://master--fruktkartan.netlify.app/ again  -- it looks like it works. (but it seems to be configured to use Nodejs 18.x -- tell me if I should change that.

That's running towards our live db, not sure how it has been configured for that, but it seems to work. 

Our Netlify setup only builds this frontend repo. Could be complicated to do more, and Netlify is a bit convoluted to me... But I have barely used it.

I've arrange so that netlify also builds the `vue` branch, and as well `netlify/*` branches. The latter should appear at `https://netlify-FOO--fruktkartan.netlify.app`

**Comment by rotsee** [comment]
@quite  I'd suggest using settting the Node version to 22.x. That is the latest LTS release (and what I use at home).

**Comment by rotsee** [comment]
@quite 
As far as I can tell, the Nuxt3 branch is now feature complete. Should we merge?

**Comment by quite** [comment]
Yes maybe let's just go for it. I pushed a few small things just now to `vue3`, that I previously committed to `master` after you branched it off to `vue3` (basically `git log base-vue3...master`).

The `vue3` branch does not merge nicely into `master`. Maybe it would be a small matter by force-pushing `vue3` to `master`. Not the done thing normally, but hey...

**Comment by quite** [comment]
I'll do it then!

**Comment by quite** [comment]
🎉🎉🎉
